### ISSUE TITLE
Proposal of fixing missing Findgtest.cmake warning in clear_costmap_recovery and costmap_2d

### ISSUE THREAD

**Issue by arne48** [ros-planning/navigation#242]
For solving this warning I would propose:
To change the "find_package(gtest)" instruction in the CMakeLists.txt of  
clear_costmap_recovery in line 43 
and costmap_2d in line 143 
to  "find_package(GTest)". 
This will lead to bigger problems when testing is intensified in the future.
So hopefully this little typo can be fixed quite soon.

**Comment by DLu** [comment]
Thanks for the suggested fix. It would be good to fix this. However, on my machine, changing it to `find_package(GTest REQUIRED)` results in the following. 

```
-- costmap_2d: 1 messages, 0 services
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:91 (MESSAGE):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:252 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-2.8/Modules/FindGTest.cmake:150 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  navigation/costmap_2d/CMakeLists.txt:143 (find_package)


-- Configuring incomplete, errors occurred!
make: *** [cmake_check_build_system] Error 1
Invoking "make cmake_check_build_system" failed
```

(Changing it without the REQUIRED results in different errors.)

I am not a GTest/CMake expert. Do you know what the problem might be?

(my repo: https://github.com/DLu/navigation/commit/feae132229a1ef6674f95474f49d75f670d2cf93)

**Comment by arne48** [comment]
To me this looks like the GTest library isn't installed correctly on your system. 
While preparing GTest I hat the problem that on Ubuntu there is no precompiled library available.
So at first I installed the "libgtest-dev" package from the repository  and then did this to build and put the library in place:
cd /usr/src/gtest
sudo cmake .
sudo make
sudo cp libg\* /usr/lib/
Maybe you can try this out and report if this solved your problem. I was trying some other things on the tests and they are running through now.

**Comment by DLu** [comment]
I figured it was something like that. That seemed to solve the problem, although I can't imagine why there's no precompiled library.

**Comment by mikeferguson** [comment]
@dirk-thomas Do you happen to know where does gtest come from? I notice in ros_comm that catkin_add_gtest is used all over, but I can't find a "find_package(GTest)" or similar anywhere -- is gtest built within rostest/rosunit or something??

I'm highly confused here, because our gtests do work -- but we never find gtest as a package...?

**Comment by dirk-thomas** [comment]
Since using GTest is not that straight forward (you have to build the library from source on demand) `catkin` provides that logic out-of-the-box: https://github.com/ros/catkin/blob/c58b1d7f16beaa279d9ae4ab019a2c34ad70c63e/cmake/test/gtest.cmake#L61

Whenever a package uses `catkin_add_gtest` that happens behind the scene.

**Comment by mikeferguson** [comment]
thanks @dirk-thomas

**Comment by mikeferguson** [comment]
Fixed by #252
